java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/CompactSortedFiles.java

```diff
@@ -187,20 +192,38 @@ private Configuration getConfiguration() {
     public static void updateStateStoreSuccess(
             CompactionJob job,
             long recordsWritten,
-            StateStore stateStore) throws StateStoreException {
+            StateStore stateStore) throws StateStoreException, InterruptedException {
+        updateStateStoreSuccess(job, recordsWritten, stateStore,
+                JOB_ASSIGNMENT_WAIT_ATTEMPTS,
+                new ExponentialBackoffWithJitter(JOB_ASSIGNMENT_WAIT_RANGE));
+    }
+
+    public static void updateStateStoreSuccess(
+            CompactionJob job,
+            long recordsWritten,
+            StateStore stateStore,
+            int jobAssignmentWaitAttempts,
+            ExponentialBackoffWithJitter jobAssignmentWaitBackoff) throws StateStoreException, InterruptedException {
         FileReference fileReference = FileReference.builder()
                 .filename(job.getOutputFile())
                 .partitionId(job.getPartitionId())
                 .numberOfRecords(recordsWritten)
                 .countApproximate(false)
                 .onlyContainsDataForThisPartition(true)
                 .build();
-        try {
-            stateStore.atomicallyReplaceFileReferencesWithNewOne(job.getId(), job.getPartitionId(), job.getInputFiles(), fileReference);
-            LOGGER.debug("Updated file references in state store");
-        } catch (StateStoreException e) {
-            LOGGER.error("Exception updating StateStore (moving input files to ready for GC and creating new active file): {}", e.getMessage());
-            throw e;
+        FileReferenceNotAssignedToJobException failure = null;
+        for (int attempts = 0; attempts < jobAssignmentWaitAttempts; attempts++) {
+            jobAssignmentWaitBackoff.waitBeforeAttempt(attempts);
+            try {
+                stateStore.atomicallyReplaceFileReferencesWithNewOne(job.getId(), job.getPartitionId(), job.getInputFiles(), fileReference);
+                LOGGER.debug("Updated file references in state store");
+                return;
+            } catch (FileReferenceNotAssignedToJobException e) {
+                LOGGER.warn("Job not yet assigned to input files, {} attempts of {}: {}",
+                        attempts, jobAssignmentWaitAttempts, e.getMessage());
```

Comment:
The log message here reads a bit strange: `2 attempts of 3` as an example. Perhaps `Attempt 2 of 3` would read better?

Reply:
Thanks, fixed